Test1/src/leetcode/LeetCode15.java

```diff
@@ -0,0 +1,51 @@
+package leetcode;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * https://leetcode.com/problems/3sum/ 
+ * @author minibig
+ */
+public class LeetCode15 {
+	
+	public List<List<Integer>> threeSum(int[] nums) {
+		List<List<Integer>> resultAllList = new ArrayList<List<Integer>>();
+
+		Arrays.sort(nums);
+		
+		int pick = 0;
+		int target = 0;
+        int rIdx = nums.length - 1;
+		for(int i = 0; i < nums.length - 2; i++) {
+			pick = nums[i];
+			target = 0 - pick;  // pick + target = 0
+			
+			int sum = 0;
+			int l = i + 1;
+			int r = rIdx;
+			while(l < r) {
+				sum = nums[l] + nums[r];
+				if(sum > target) {
+					r--;
+				}else if(sum < target) {
+					l++;
+				}else { // sum == target
+					List<Integer> list = new ArrayList<Integer>();
+					list.add(nums[i]);
+					list.add(nums[l]);
+					list.add(nums[r]);
+					resultAllList.add(list);
+					if(nums[i] == nums[l] && nums[l] == nums[r]) {
+						rIdx--;
+					}
```

Comment:
I don't understand why the condition is like this. Can you explain your thought of this if-condition?

Reply:
I was also thinking that I thought wrong. 
I just solved so many only zeros case by using this. 
But i think this is the wrong way.

Reply:
only 3 zeros case pass this if phrase. 
what i thought at first was, if I delete one zero from the main arr, maybe it will reduce finding every case.
Then for this, I would add set that have index of one zero, and if set has that index pass it.
how about this?

Reply:
I think the problem is not about the 3 zeroes cases.

What you got was a TLE.

![image](https://user-images.githubusercontent.com/7591613/190891237-1f9cafd9-e8b8-4648-a66a-95e6cf16f757.png)

You can click "View all" to see the full test case.

You need to think carefully about why your code got TLE.